File: smk_scm/public/js/request_for_quotation.js
frappe.ui.form.on('Request for Quotation', {
    on_submit(frm) {
        let rfq_details = `
            <table border="1" cellpadding="5" cellspacing="0">
                <thead>
                    <tr>
                        <th>Item Name</th>
                        <th>Quantity</th>
                        <th>UOM</th>
                        <th>Required By</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
        `;
        frm.doc.items.forEach(item => {
            let descriptionText = item.description.replace(/<\/?p[^>]*>/g, '').trim();
            var inputDate = item.schedule_date;
            var parts = inputDate.split("-");
            var year = parts[0];
            var month = parts[1];
            var day = parts[2];
            var formattedDate = day ? `${day}-${month}-${year}` : `${year}`;
            rfq_details += `
                <tr>
                    <td>${item.item_name}</td>
                    <td>${item.qty}</td>
                    <td>${item.uom}</td>
                    <td>${formattedDate}</td>
                    <td>${descriptionText}</td>
                </tr>
            `;
        });
        rfq_details += `
                </tbody>
            </table>
        `;

        let terms = frm.doc.terms.replace(/<\/?[^>]+(>|$)/g, '').trim();
        frm.doc.suppliers.forEach(supplier => {
            frappe.call({
                method: 'smk_scm.public.py.request_for_quotation.send_email',
                args: {
                    name: frm.doc.name,
                    company: frm.doc.company,
                    recipient_id: supplier.email_id,
                    recipient: supplier.supplier,
                    rfq_details,
                    tc_name : frm.doc.tc_name,
                    terms
                },
                callback: function(response) {
                    if (response.message) {
                        frappe.msgprint('An Email sent successfully');
                    }
                }
            });
        });
    }
});

File: smk_scm/public/py/request_for_quotation.py
import frappe
from frappe.email.queue import flush
from frappe.utils import get_url

@frappe.whitelist()
def send_email(name, company, recipient_id, recipient, rfq_details, tc_name, terms):
    logo_url = get_url("/files/SMK logo.jpg")
    message = f"""
    <p>Dear {recipient},</p>
    <p>I hope this message finds you well.
    <br>We are currently in the process of reviewing our suppliers and would like to request a quotation for the following items. Please provide us with the item rates, including any applicable taxes, delivery charges, and terms of payment.</p>
    <b>Item Details:</b>
    <p>{rfq_details}</p>
    """
    if tc_name:
        message += f"<p><b>Terms: {tc_name}</b></p>"
    if terms:
        message += f"<p><b>Terms and Conditions:</b> {terms}</p>"
    message += f"""
    <p>Please provide us with a detailed quotation, and do not hesitate to contact me if you require any further information.
    <br>Thank you for your attention to this request. We look forward to your prompt response.</p>
    <p>Best regards,<br>{company}</p>
    <p><img src="{logo_url}" alt="SMK Petrochemicals" width="200" /></p>
    """
    frappe.sendmail(
        recipients=[recipient_id],
        subject="Request for Quotation for Items",
        message=message
    )
    flush()